cli/src/pcluster/config/cluster_config.py

```diff
@@ -1522,6 +1564,37 @@ def _register_validators(self):
                     EfaOsArchitectureValidator,
                     efa_enabled=compute_resource.efa.enabled,
                     os=self.image.os,
-                    # FIXME: head_node.architecture vs compute_resource.architecture?
                     architecture=self.head_node.architecture,
                 )
+                self._register_validator(
+                    ComputeAMIOsCompatibleValidator,
+                    os=self.image.os,
+                    image_id=self.image_dict[queue.name],
+                )
+
+    @property
+    def image_dict(self):
+        """Return image dict of queues, key is queue name, value is image id."""
+        if self.__image_dict:
+            return self.__image_dict
+        self.__image_dict = {}
+
+        for queue in self.scheduling.queues:
+            if queue.queue_ami_id:
```

Comment:
In this case where the ami is a string and not a boolean you can do:
```python
self.__image_dict[queue.name] = queue.queue_ami_id or self.image.custom_ami or self.official_ami
```

Reply:
done